api/v1beta2/configuration_types.go

```diff
@@ -96,12 +98,25 @@ type Property struct {
 	Value string `json:"value,omitempty"`
 }
 
-// Backend stores the state in a Kubernetes secret with locking done using a Lease resource.
+// Backend describes the Terraform backend configuration
 type Backend struct {
 	// SecretSuffix used when creating secrets. Secrets will be named in the format: tfstate-{workspace}-{secretSuffix}
+	// Deprecated: use the `type` and `config` instead
 	SecretSuffix string `json:"secretSuffix,omitempty"`
 	// InClusterConfig Used to authenticate to the cluster from inside a pod. Only `true` is allowed
+	// Deprecated: use the `type` and `config` instead
 	InClusterConfig bool `json:"inClusterConfig,omitempty"`
+
+	// HCL allows users to use raw hcl code to specify their Terraform backend configuration
+	HCL string `json:"hcl,omitempty"`
+
+	// Type specifies the Terraform backend type, for example, "kubernetes", "s3", etc
+	Type string `json:"type,omitempty"`
+	// Config is the detail configurations of the Terraform backend,
+	// and it represents the key-value pairs inside the terraform backend block in the hcl files
```

Comment:
Users don't know how to set such fields. As Terraform backends are limited, we can support them one by one.

Reply:
Do you mean that the users can write the `spec.backend` like this?

```yaml
spec:
  backend: 
    s3:
      bucket: "mybucket"
      key: "path/to/my/key"
      region: "my-region"
```

Reply:
On the one hand, we need to use have a clear guide on how to set the backend just per the API, on the other hand, the `spec.backend` should be as simple as possible. Let's set aside some time on the design to check whether it's possible.

Maybe we can refer a Backend object, like we referring to a Provider object in Configuration.

Reply:
> Maybe we can refer a Backend object, like we referring to a Provider object in Configuration.

I don't think it's necessary to maintain an additional `Backend` object, if only to provide custom backend configuration in the `Configuration` object. 

But if we can add `backup and restore` configuration and `backup` life cycle management to `Backend` object, I think it will make sense.

Reply:
+1.